com.unity.netcode.gameobjects/Tests/Editor/Messaging/MessageRegistrationTests.cs

```diff
@@ -179,5 +179,92 @@ public void WhenCreatingMessageSystem_BoundTypeMessageHandlersAreRegistered()
                 Assert.AreEqual(handlerFour, systemThree.MessageHandlers[systemThree.GetMessageType(typeof(TestMessageFour))]);
             }
         }
+
+        internal class AAAEarlyLexicographicNetworkMessage : INetworkMessage
+        {
+            public void Serialize(FastBufferWriter writer)
+            {
+            }
+
+            public bool Deserialize(FastBufferReader reader, ref NetworkContext context)
+            {
+                return true;
+            }
+
+            public void Handle(ref NetworkContext context)
+            {
+            }
+        }
+
+#pragma warning disable IDE1006
+        internal class zzzLateLexicographicNetworkMessage : AAAEarlyLexicographicNetworkMessage
+        {
+        }
+#pragma warning restore IDE1006
+
+        internal class OrderingMessageProvider : IMessageProvider
+        {
+            public List<MessagingSystem.MessageWithHandler> GetMessages()
+            {
+                var listMessages = new List<MessagingSystem.MessageWithHandler>();
+
+                var messageWithHandler = new MessagingSystem.MessageWithHandler();
+
+                messageWithHandler.MessageType = typeof(zzzLateLexicographicNetworkMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(ConnectionRequestMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(ConnectionApprovedMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(OrderingMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(AAAEarlyLexicographicNetworkMessage);
+                listMessages.Add(messageWithHandler);
+
+                return listMessages;
+            }
+        }
+
+        [Test]
+        public void MessagesGetPrioritizedCorrectly()
+        {
+            var sender = new NopMessageSender();
+            var provider = new OrderingMessageProvider();
+            var messagingSystem = new MessagingSystem(sender, null, provider);
+
+            // the 3 priority messages should appear first, in lexicographic order
+            Assert.AreEqual(messagingSystem.MessageTypes[0], typeof(ConnectionApprovedMessage));
+            Assert.AreEqual(messagingSystem.MessageTypes[1], typeof(ConnectionRequestMessage));
+            Assert.AreEqual(messagingSystem.MessageTypes[2], typeof(OrderingMessage));
+
+            // the other should follow after
+            Assert.AreEqual(messagingSystem.MessageTypes[3], typeof(AAAEarlyLexicographicNetworkMessage));
+            Assert.AreEqual(messagingSystem.MessageTypes[4], typeof(zzzLateLexicographicNetworkMessage));
+
+            // there should not be any extras
+            Assert.AreEqual(messagingSystem.MessageHandlerCount, 5);
+            Assert.AreEqual(messagingSystem.MessageHandlerCount, 5);
```

Comment:
copy paste error here and lines 265-266? You're asserting the same thing twice in a row...

Reply:
Good catch. Initially, this tested the size of both arrays (handlers and types), but then I realized `MessageHandlerCount` was the actual size of both array. So I (silly) just changed both check to use `MessageHandlerCount`, resulting in duplicate checks. :-)

Will fix.